pyccel/codegen/printing/ccode.py

```diff
@@ -431,6 +431,12 @@ def fill_NumpyArange(self, expr, lhs):
 
     # ============ Elements ============ #
 
+    def _print_PythonAbs(self, expr):
+        """ print the python builtin function abs
+        args : variable
+        """
+        return "abs({})".format(self._print(expr.arg))
```

Comment:
Not like python C ```abs``` work only with integer types ```int abs(int x)```
so if you try  : 
```PYTHON
a = abs(2.2)
print(a)
```
you get ```2.000000```
try to use the corresponding absolute value for each type : (abs for int) and (fabs for float) etc ...
you may need to add import  like <math.h>  or  <complex.h> you can do that by using 
same as here https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/ccode.py#L459

Reply:
you can get the type of the arguments in ```expr.arguments.dtype``` see  ``` pyccel/ast/datatypes``` for classes definitions